core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -22,6 +22,8 @@ import org.apache.kafka.common.TopicPartition
 
 import scala.collection.{Map, Seq, Set, mutable}
 
+case class PartitionReplicaAssignment(replicas: Seq[Int], addingReplicas: Seq[Int], removingReplicas: Seq[Int])
```

Comment:
I am not sure where to best place this case class. Considering that it will be returned by our `KafkaZkClient`, some systems that are depending on this public (but undocumented) API might want to use the class